bigchaindb/core.py

```diff
@@ -684,9 +621,10 @@ def get_last_voted_block(self):
 
         except r.ReqlNonExistenceError:
             # return last vote if last vote exists else return Genesis block
-            return list(self.connection.run(
+            block = list(self.connection.run(
```

Comment:
I think this is really misleading; it'd be nicer to split these out into two statements (assuming we really have to cast the rdb result to a list first):

```python
res = self.connection.run(
    r.table('bigchain', read_mode=self.read_mode)
    .filter(util.is_genesis_block))
block = list(res)[0]
```

`block = list(...)` and then `Block.from_dict(block)` really makes you think how `from_dict` works with a list, until you see the trailing `[0]` at the end.